webflow-hacks/form-hacks/43/43.ts

```diff
@@ -0,0 +1,43 @@
+// when dom is ready
+document.addEventListener('DOMContentLoaded', function () {
+  const TOTAL_SELECTOR = '[fs-hacks-element="total-value"]';
+  const HIDDEN_INPUT_SELECTOR = '[fs-hacks-element="hidden-total"]';
+  const ADD_VALUE_ATTRIBUTE = 'add-value';
+  // get all elements
+  const totalValueDiv = document.querySelector<HTMLDivElement>(TOTAL_SELECTOR);
+  const hiddenTotalInput = document.querySelector<HTMLInputElement>(HIDDEN_INPUT_SELECTOR);
+  const radios = document.querySelectorAll<HTMLInputElement>(`[${ADD_VALUE_ATTRIBUTE}]`);
+  if (!totalValueDiv || !hiddenTotalInput) return;
+  let brandingTotal = 0;
+  let developmentTotal = 0;
+  radios.forEach((radio) => {
+    // listen to the radio.
+    radio.addEventListener('input', function () {
+      if (!radio.checked) return;
+      // find if element is in the branding or development div
+      if (radio.getAttribute('name') === 'Branding') {
+        // get the value of the radio
+        brandingTotal = Number(radio.getAttribute(ADD_VALUE_ATTRIBUTE));
+      }
+      if (radio.getAttribute('name') === 'Development') {
+        developmentTotal = Number(radio.getAttribute(ADD_VALUE_ATTRIBUTE));
+      }
+      const total = brandingTotal + developmentTotal;
+      updateTotals(total, totalValueDiv, hiddenTotalInput);
+    });
+  });
+});
```

Comment:
This could still be simplified even more.
We know that the radio groups that we're dealing with are `Branding` and `Development`, se we can just query the radios using both group names.
As for the values, there's no need to have an `ADD_VALUE_ATTRIBUTE`. We can just use the native `value` HTML attribute.

```typescript
document.addEventListener('DOMContentLoaded', function () {
  const TOTAL_SELECTOR = '[fs-hacks-element="total-value"]';
  const HIDDEN_INPUT_SELECTOR = '[fs-hacks-element="hidden-total"]';
  const BRANDING_GROUP_NAME = 'Branding';
  const DEVELOPMENT_GROUP_NAME = 'Branding';

  // get all elements
  const totalValueDiv = document.querySelector<HTMLDivElement>(TOTAL_SELECTOR);
  const hiddenTotalInput = document.querySelector<HTMLInputElement>(HIDDEN_INPUT_SELECTOR);
  const radios = document.querySelectorAll<HTMLInputElement>(
    `input[name="${BRANDING_GROUP_NAME}"], input[name="${DEVELOPMENT_GROUP_NAME}"]`
  );
  if (!totalValueDiv || !hiddenTotalInput) return;

  let brandingTotal = 0;
  let developmentTotal = 0;

  radios.forEach((radio) => {
    // listen to the radio.
    radio.addEventListener('input', function () {
      const { name, value, checked } = radio;

      if (!checked) return;

      if (name === BRANDING_GROUP_NAME) {
        brandingTotal = Number(value);
      }

      if (name === DEVELOPMENT_GROUP_NAME) {
        developmentTotal = Number(value);
      }

      const total = brandingTotal + developmentTotal;
      updateTotals(total, totalValueDiv, hiddenTotalInput);
    });
  });
});
```

Reply:
awesome

Reply:
I have added it to the comments on top of the file